plugins/out_oci_logan/oci_logan.c

```diff
@@ -0,0 +1,1327 @@
+    if (!error_response) {
+        flb_errno();
+        flb_free(tokens);
+        return NULL;
+    }
+
+    /* Parse JSON tokens */
+    for (i = 0; i < tok_size; i++) {
+        t = &tokens[i];
+
+        if (t->start == -1 || t->end == -1 || (t->start == 0 && t->end == 0)) {
+            break;
+        }
+
+        if (t->type != JSMN_STRING) {
+            continue;
+        }
+
+        key = response + t->start;
+        key_len = (t->end - t->start);
+
+        i++;
+        t = &tokens[i];
+        val = response + t->start;
+        val_len = (t->end - t->start);
+
+        if (val_len < 1) {
+            continue;
+        }
+
+        if ((key_len == sizeof(FLB_OCI_ERROR_RESPONSE_CODE) - 1)
+            && strncasecmp(key, FLB_OCI_ERROR_RESPONSE_CODE,
+                           sizeof(FLB_OCI_ERROR_RESPONSE_CODE) - 1) == 0) {
+            // code
+            error_response->code = flb_sds_create_len(val, val_len);
+            if (!error_response->code) {
+                flb_free(error_response);
+                flb_free(tokens);
+                return NULL;
+            }
+        }
+        else if ((key_len == sizeof(FLB_OCI_ERROR_RESPONSE_MESSAGE) - 1)
+            && strncasecmp(key, FLB_OCI_ERROR_RESPONSE_MESSAGE,
+                           sizeof(FLB_OCI_ERROR_RESPONSE_MESSAGE) - 1) == 0) {
+
+            // message
+            error_response->message = flb_sds_create_len(val, val_len);
+            if (!error_response->message) {
+                flb_free(error_response);
+                flb_free(tokens);
+                return NULL;
+            }
+        }
+    }
+
+    flb_free(tokens);
+    return error_response;
+}
+
+static int retry_error(struct flb_http_client *c, struct flb_oci_logan *ctx)
+{
+    struct flb_oci_error_response *error_response = NULL;
+    int tmp_len;
+    int ret = FLB_FALSE;
+
+    // possible retry error message
+    if ( !(c->resp.status == 400 || c->resp.status == 401
+        || c->resp.status == 404 || c->resp.status == 409
+        || c->resp.status == 429 || c->resp.status == 500)) {
+        return FLB_FALSE;
+    }
+
+    // parse error message
+    error_response = parse_response_error(ctx, c->resp.payload,
+                                          c->resp.payload_size);
+    if (!error_response) {
+        return FLB_FALSE;
+    }
+
+    if (error_response->code) {
+        tmp_len = flb_sds_len(error_response->code);
+        if (c->resp.status == 400
+            && (tmp_len == sizeof(FLB_OCI_ERROR_CODE_RELATED_RESOURCE_NOT_FOUND) - 1)
+            && strncasecmp(error_response->code, FLB_OCI_ERROR_CODE_RELATED_RESOURCE_NOT_FOUND, tmp_len) == 0) {
+            ret = FLB_TRUE;
+        } else if( c->resp.status == 401
+            &&( tmp_len == sizeof(FLB_OCI_ERROR_CODE_NOT_AUTHENTICATED)-1 )
+            && strncasecmp(error_response->code, FLB_OCI_ERROR_CODE_NOT_AUTHENTICATED, tmp_len) == 0) {
+            ret = FLB_TRUE;
+        } else if (c->resp.status == 404
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions